### PR TITLE
Fix append capability after geometry refactoring.

### LINKED ISSUE #206: Appending to an existing vector layer is broken by the geometry refactoring.
The "append" functionality no longer works in OpenSpaceNet. The signs point to the resent `FeatureSet` refactoring.

### Steps to Reproduce

Run OSN with the following parameters:

```
OpenSpaceNet detect --model detectnet.gbdxm --output detectnet_detection.shp --nms 60 --step-size 512 --confidence 50 --append --image 055675456010_01_assembly.tif
```

Note that the model, input and output locations, and the detection parameters can be different, the important part is the `--append` flag.

### What Should Happen

If `detectnet_detection.shp` exists, OSN will append detected features to this same file/layer. If it doesn't exist, it should create a new file/layer.

### What Actually Happens

OSN Displays the following error:
```
   ____                    _____                      _   _      _
  / __ \                  / ____|                    | \ | |    | |
 | |  | |_ __   ___ _ __ | (___  _ __   __ _  ___ ___|  \| | ___| |_
 | |  | | '_ \ / _ \ '_ \ \___ \| '_ \ / _` |/ __/ _ \ . ` |/ _ \ __|
 | |__| | |_) |  __/ | | |____) | |_) | (_| | (_|  __/ |\  |  __/ |_ _ _ _
  \____/| .__/ \___|_| |_|_____/| .__/ \__,_|\___\___|_| \_|\___|\__(_|_|_)
        | |                     | |
        |_|                     |_|

Version: 0.3.0+29
DeepCore Version: 0.1.1+247

Opening image...
Reading model package...
Reading model...
Note that future Caffe releases will only support input layers and not input fields.

Model Name: Airplane; Version: 1.0; Created: 2017-Jan-25 22:54:19
Description: DetectNet model for airplane detection
Dimensions (pixels): (768, 768); Color Mode: RGB; Image Type: jpg
Bounding box (lat/lon): (-180, -90) : (90, 180)
Labels: Airplane

Initializing the output feature set...
ERROR 1: Layer 'detectnet_detection' already exists
Error creating a vector layer: Layer 'detectnet_detection' already exists
```

### Impact

OSN can no longer append multiple detection runs to the same feature set.

### PR DESCRIPTION
**Pull request by LetThereBeDwight**
resolve #206

### CODE REVIEW

File: src/OpenSpaceNet.cpp [hunk 1]
```diff
@@ -276,7 +276,12 @@ void OpenSpaceNet::initFeatureSet()
     VectorOpenMode openMode = args_.append ? APPEND : OVERWRITE;
 
     featureSet_ = make_unique<FeatureSet>(args_.outputPath, args_.outputFormat, openMode);
-    layer_ = featureSet_->createLayer(args_.layerName, sr_, args_.geometryType, definitions);
+
+    if (openMode == OVERWRITE) {
+        layer_ = featureSet_->createLayer(args_.layerName, sr_, args_.geometryType, definitions);    
+    } else if (openMode == APPEND) {
```
Comment (avitebskiy) [inline src/OpenSpaceNet.cpp]: You want to make sure the layer is there first, since "append" is allowed to create an new layer if one doesn't exist.

**Comment by avitebskiy** [comment]
test this please

**Comment by avitebskiy** [comment]
test this please

**Comment by avitebskiy** [comment]
test this please

**Comment by geovations** [comment]
test this please

**Pull request merged**